WordPressComStatsiOS/WordPressComStatsiOS/UI/WPStatsGraphViewController.m

```diff
@@ -208,6 +217,50 @@ - (void)setVisits:(StatsVisits *)visits forSummaryType:(StatsSummaryType)summary
 
 #pragma mark - Private methods
 
+- (CGFloat)calculateDateFontSize
+{
+    // step through each of the dates
+    // and calculate what font size each would actually be displayed at
+    // then return the smallest font size out of all of them
+    CGFloat smallestFontSize = [[WPStyleGuide axisLabelFont] pointSize];
+
+    for (StatsSummary *summary in self.statsData) {
+        NSInteger index = (NSInteger)[self.statsData indexOfObject:summary];
+        NSIndexPath *indexPath = [NSIndexPath indexPathForItem:index inSection:0];
+
+        WPStatsGraphBarCell *cell = [self collectionView:self.collectionView cellForItemAtIndexPath:indexPath];
+        CGFloat currentFontSize = [self fontSizeForString:cell.barName inRect:cell.frame];
+
+        if (index == 0) {
+            smallestFontSize = currentFontSize;
+        } else if (currentFontSize < smallestFontSize) {
+            smallestFontSize = currentFontSize;
+        }
+    }
+
+    return smallestFontSize;
+}
+
+- (CGFloat)fontSizeForString: (NSString *)string inRect: (CGRect)rect
+{
+    UILabel *label = [[UILabel alloc] initWithFrame:CGRectMake(0, 0, CGRectGetWidth(rect), 17.0f)];
```

Comment:
@jim-rhoades - I noticed that the label may not really be necessary in this method.  Is there any specific reason to have it?

Reply:
It's used by the rest of the method to calculate the font size... but honestly I wasn't in love with allocating temporary UILabel's just to figure out what the smallest font size should be. I was rushed and had family in town the day I was playing around with that, so didn't have time to figure out if there's a better way. I'll look at it again.